app/controllers/personal_detail_controller.rb

```diff
@@ -11,7 +13,9 @@ def update
     @step = step_class.new(step_params)
 
     if @step.valid?
-      current_app.update!(step_params)
+      app = current_or_new_snap_application
+      app.update!(step_params)
+      set_current_snap_application(app)
```

Comment:
I'm guessing we'll be doing this a bunch - what do you think about some kind of `SnapApplicationUpdater.new(step_params).update_and_set_current` collaborator? Premature?